[export] refactor _Dim into Dim

pianpwk opened the pull request:
Summary: forward fix T218515233

Test Plan: test_export

Differential Revision: D71769231




cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

jingsh commented on torch/export/dynamic_shapes.py:
```diff
@@ -70,76 +70,88 @@ def STATIC():
-    def _derive(cls, fn):
-        return _DerivedDim(cls._derived_name(fn), (int,), {"root": cls, "fn": fn})
+    def _derive(self, fn) -> "Dim":
+        return _DerivedDim(self._derived_name(fn), (int,), {"root": self, "fn": fn})
```
this error is legit, can you fix this?

pianpwk committed:
[export] refactor _Dim into Dim (#149891)

Summary:

forward fix T218515233

Test Plan: test_export

Differential Revision: D71769231

titaiwangms commented:
Changed the release note from onnx to export
